Compilation fails on MSYS2

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Affected Build(s)

All recent

### Description of Issue

When following the steps in the official Yuzu Windows MSYS2 compilation howto there are 2 build errors that are preventing the compilation to succeed.

```
C:/Users/Philipp/git/yuzu-mainline/src/common/fs/file.h:40:25: error: no matching function for call to 'std::basic_ifstream<char>::open(std::u8string, std::ios_base::openmode&)'
   40 |         file_stream.open(ToU8String(path), open_mode);
```

and 

```
C:/msys64/mingw64/include/c++/13.2.0/fstream:988:9: error: 'class std::__cxx11::basic_string<char8_t>' has no member named 'make_preferred'
```

When following the official howto, GCC for Windows is being used in MSYS2. To me it appears this makes compilation following a different path compared with using MSVC. On Linux the invocation of ToU8String(path) appearently is not needed which probably is the reason why GCC is working there.
Please fix this issue so we can get more optimized builds on Windows.

### Expected Behavior

The build finishing without errors.

### Reproduction Steps

Trying to build on Windows using MSYS2.

### Log File

Please find the relevant parts above (issue description).

### System Configuration

CPU: irrelevant
GPU/Driver: irrelevant
RAM: irrelevant
OS: Windows 10

--- a/src/common/fs/file.h
+++ b/src/common/fs/file.h
@@ -37,7 +37,7 @@ void OpenFileStream(FileStream& file_stream, const std::filesystem::path& path,
 template <typename FileStream, typename Path>
 void OpenFileStream(FileStream& file_stream, const Path& path, std::ios_base::openmode open_mode) {
     if constexpr (IsChar<typename Path::value_type>) {
-        file_stream.open(ToU8String(path), open_mode);
+        file_stream.open(std::filesystem::path{ToU8String(path)}, open_mode);
     } else {
         file_stream.open(std::filesystem::path{path}, open_mode);
     }

